internal/testutils/testcluster.go

```diff
@@ -683,35 +683,64 @@ func EventuallyDpuOperatorConfigReady(c client.Client, logger logr.Logger, cr *c
 	return dpuOperatorConfig
 }
 
-func EventuallyDpuOperatorConfigDeleted(c client.Client, name string, namespace string, timeout time.Duration, interval time.Duration) {
+func EventuallyNoDpuOperatorConfig(c client.Client, timeout time.Duration, interval time.Duration) {
 	onFailure := func() {
-		fmt.Println(LogDpuOperatorConfigDiagnostics(c, name))
+		crList := &configv1.DpuOperatorConfigList{}
+		err := c.List(context.Background(), crList)
+		if err != nil {
+			fmt.Printf("Failed to list DpuOperatorConfigs for diagnostics: %v\n", err)
+			return
+		}
+		if len(crList.Items) > 0 {
+			fmt.Printf("Found %d DpuOperatorConfig CRs:\n", len(crList.Items))
+			for _, cr := range crList.Items {
+				fmt.Printf("  - %s/%s", cr.Namespace, cr.Name)
+				if len(cr.Finalizers) > 0 {
+					fmt.Printf(" (finalizers: %v)", cr.Finalizers)
+				}
+				if cr.DeletionTimestamp != nil {
+					fmt.Printf(" (deletion started: %v)", cr.DeletionTimestamp)
+				}
+				fmt.Printf("\n")
+				fmt.Println(LogDpuOperatorConfigDiagnostics(c, cr.Name))
+			}
+		}
 	}
 
 	startTime := time.Now()
 
 	AssertEventually(
 		func() error {
-			dpuConfig := &configv1.DpuOperatorConfig{}
-			err := c.Get(context.Background(), types.NamespacedName{
-				Name:      name,
-				Namespace: namespace,
-			}, dpuConfig)
-			if errors.IsNotFound(err) {
-				return nil
-			}
+			crList := &configv1.DpuOperatorConfigList{}
+			err := c.List(context.Background(), crList)
 			if err != nil {
-				return fmt.Errorf("error checking DpuOperatorConfig %s in %s: %v", name, namespace, err)
+				return fmt.Errorf("failed to list DpuOperatorConfigs: %v", err)
+			}
+
+			if len(crList.Items) > 0 {
```

Comment:
Duplication of come from lines 694-706. We are just checking the finalizer label for the CR right? And reporting it?